stepflow-rs/crates/stepflow-main/src/visualize.rs

```diff
@@ -0,0 +1,440 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements.
+// See the NOTICE file distributed with this work for additional information regarding copyright
+// ownership.  The ASF licenses this file to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance with the License.  You may obtain a
+// copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.  See the License for the specific language governing permissions and limitations under
+// the License.
+
+use std::collections::HashMap;
+use std::fmt::Write as _;
+use std::path::Path;
+
+use error_stack::{report, ResultExt as _};
+use stepflow_analysis::{Dependency, analyze_flow_dependencies, FlowAnalysis};
+use stepflow_core::workflow::Flow;
+use stepflow_plugin::routing::PluginRouter;
+
+use crate::error::{MainError, Result};
+
+/// Output formats supported for visualization
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum OutputFormat {
+    /// DOT graph format
+    Dot,
+    /// SVG image format
+    Svg,
+    /// PNG image format  
+    Png,
+}
+
+impl std::str::FromStr for OutputFormat {
+    type Err = String;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
```

Comment:
nit: Maybe add trim on here for funzies/completeness?

Reply:
Also, in doing a small deep dive on enums and why there is no valueOf equivalent (totally get it now), I learned about the `strum` crate which would allow something like the following:
```
use strum_macros::{Display, EnumString};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Display, EnumString)]
#[strum(serialize_all = "lowercase")]
pub enum OutputFormat {
    Dot,
    Svg,
    Png,
}

// Now you get both for free:
let png = OutputFormat::Png;
println!("{}", png);  // prints: "png"
let parsed: OutputFormat = "png".parse().unwrap();  // works!
```